### ISSUE TITLE
xorg.xkeyboardconfig: altwin:swap_ralt_rwin option not built or working

### ISSUE THREAD

**Issue by dyfrgi** [NixOS/nixpkgs#326154]
### Describe the bug
When using `setxkbmap -option altwin:swap_ralt_rwin`, no option is applied.

### Steps To Reproduce
1. Install X11 and start a session
2. run `setxkbmap -option altwin:swap_ralt_rwin` in a terminal
3. Observe that the keymap isn't changed

### Expected behavior
Alt_R and Super_R should be swapped.

### Additional context
I am not sure if this bug is properly located here or if it should go upstream. I think that the problem is that the rule exists but is not built - if you look at `symbols/altwin`, you'll see the rule for `swap_ralt_rwin`, but that rule doesn't appear anywhere in the `build` directory after running the build for the `xorg.xkeyboardconfig` package.

After install, the rule is in `share/X11/xkb/symbols/altwin`, but not referenced in `share/X11/xkb/rules/evdev` or `evdev.lst`.

It looks like this option does not appear in the `rules` directory of the source which, if I understand the xkeyboard-config build correctly, is probably why it's not showing up in the rules output.

It's possible that this is in fact a feature request to add the patch which other distributions evidently have to enable this, since I see many reports online of people using this option successfully.

### Notify maintainers
@lovek323 (listed maintainer) @trofi (last person to upgrade xkeyboard-config version)

### Metadata

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.31, NixOS, 24.05 (Uakari), 24.05beta142.d12251ef6e8e`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.2`
 - channels(root): `"nixos-24.05"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by jtojnar** [comment]
Looks like only `altwin:swap_alt_win` is publicly exposed – it invokes `altwin:swap_lalt_lwin` and `altwin:swap_ralt_rwin` but those are both private. Before the combined option was introduced, `altwin:swap_lalt_lwin` was public. But it was [decided](https://www.google.com/search?hl=en&q=cache:https%3A%2F%2Fbugs.freedesktop.org%2Fshow_bug.cgi%3Fid%3D57813) during [the change](https://salsa.debian.org/xorg-team/data/xkeyboard-config/-/commit/1a751933e080eceedb68004958620b5d923355b1) that the left-specific option is not necessary, and the newly added right-specific one has the same visibility.

**Comment by dyfrgi** [comment]
Aha, thanks for finding that! I wasn't able to locate the discussion due to their old bug tracker being down yesterday. I [inquired upstream](https://gitlab.freedesktop.org/xkeyboard-config/xkeyboard-config/-/issues/474) and was told "MR welcome", so I gather that they would now accept this option, even if it adds further options.

I also looked more at how people are successfully using the symbol today and found that they're all doing it via Gnome, which must be compiling its own xkb descriptions using the symbols rather than depending on rules. Unfortunately that probably means there's no easily-incorporable pre-existing patch that adds this rule.

Personally for now, I've fixed this with qmk on one system. I probably won't pursue software configuration in NixOS further right now since I'm not running it on any systems with unmodifiable keyboard firmware. I'm going to close this in favor of the upstream bug. Now to figure out how to build xkeyboard-config from a local git repo, on NixOS... hopefully the current upstream is compatible so I can properly test it.
